### DISCUSSION TITLE
AWS Amplify Deployment : Stale Cache

### DISCUSSION THREAD

**Discussion by akil-rails** [vuejs/vitepress#1231] [Q&A]
I've had a couple of instances where after deploying vitepress on AWS amplify, I've had to clear the cache to get the latest changes. Is there any specific deployment step that needs to be executed? Don't have a reproducible sample as yet.

**Comment by szsascha** [comment]
I'm wondering why you're using AWS amplify for vitepress. You could simply use an S3 bucket. A S3 bucket should be cheaper and eliminate any cache problems. (Except for CDN caches. They should be refreshed any few hours automatically.)
About 2 weeks ago I wrote a little tutorial about this: https://github.com/szsascha/vue-s3-cdn
(Shouldn't be an advertisement but should be a cheaper and better option.)

**Reply by akil-rails** [reply]
AWS Amplify has some nice features - integrated with GitHub for CD, previews, password protection and some more. But this cache problem is troubling, any suggestions on what to look for ?

**Reply by brc-dd** [reply]
I'm not experienced with AWS, but from [their docs](https://aws.amazon.com/blogs/mobile/aws-amplify-console-supports-instant-cache-invalidation-and-delta-deployments/#:~:text=Instant%20cache%20invalidation%20with%20the%20Amplify%20Console) it appears Amplify automatically invalidates cache. The docs are not clear but it looks it doesn't work when performance mode is enabled. You need to manually adjust cache duration. Use whatever maximum is there for `/assets/*` (as they are hashed), for others I think you should stick with the default 10min.
